client/app.jsx

```diff
@@ -111,34 +111,90 @@ function App() {
               )}
         </Route>
         <Route path="/new-post">
-          <TopAppBar />
-          <NewPost />
-          <BottomNavBar />
+          {username
+            ? (
+            <div>
+              <TopAppBar />
+              <NewPost />
+              <BottomNavBar />
+            </div>
+              )
+            : (
+            <Redirect to="/auth" />
+              )}
         </Route>
         <Route path="/chat">
-          <TopAppBar />
-          <Chat />
-          <BottomNavBar />
+          {username
+            ? (
+            <div>
+              <TopAppBar />
+              <Chat />
+              <BottomNavBar />
+            </div>
+              )
+            : (
+            <Redirect to="/auth" />
+              )}
         </Route>
         <Route path="/favorites">
-          <TopAppBar />
-          <Favorites />
-          <BottomNavBar />
+          {username
+            ? (
+            <div>
+              <TopAppBar />
+              <Favorites />
+              <BottomNavBar />
+            </div>
+              )
+            : (
+            <Redirect to="/auth" />
+              )}
```

Comment:
Instead of including all of these here, you can put them into the specific page components (Favorites, Chat, etc) as a guard

```jsx
if (!username) return <Redirect to="/auth" />
```

alternatively, you could create a higher order component (HOC) to wrap all protected pages.

https://reactjs.org/docs/higher-order-components.html